concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md

```diff
@@ -0,0 +1,98 @@
+To correctly substitute displayed values with required values in a DataGrid/TreeList lookup column, follow these steps:
+
+1. Define the [lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).[dataSource](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#dataSource) property.
+
+2. Specify the [lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).[valueExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#valueExpr) property. Its value should match the [columns](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/).[dataField](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataField) property value. Refer to the following topic for detailed explanation: [lookup - API Section](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).
```

Comment:
I would say `value` and `type` should be the same because "3" and 3 technically match if you compare them like 3 == "3". So typing is important because we compare them like 3 === "3" :) Our customers often consider this as matching values especially the problem appears when values loaded from the server via `CustomStore`

Reply:
UPD: nevermind, found it's in the next topic :D